### DISCUSSION TITLE
How to install Imagemagick in my docker image

### DISCUSSION THREAD

**Discussion by yallavaibhav** [ImageMagick/ImageMagick#5770] [Help]
Hi friends,

I am currently working on Moviepy, but the videos I generate needs text file. The TextClip present in moviepy needs ImageMagick. For that I need to download it in my existing docker image.
'''
From python:3.8-slim-buster

Copy . .
Run pip install -r requirements.txt
CMD ["python3", "MyVideo.py"]

'''

Where can I add the code for installing the imagemagick, because I can't add two From's in the same docker file.

Please help me out

Thank you

**Comment by HBL001** [comment]
Sure this is straightforwards.

In your dockerfile add this line....


RUN apt-get update && apt-get install -y imagemagick

**Reply by HBL001** [reply]
Hi,
 
To test your installation works, 

1. pull your new container and get it started the normal way (docker-compose).

2. open a new terminal.  

colibri-imx8x-07021OO:~$ docker ps

CONTAINER ID        IMAGE               COMMAND                  CREATED                    STATUS              PORTS                      NAMES
ac58687d4245        clotspot/lol        "tini -v -- /opt/coa…"   17 minutes ago             Up 17 minutes       0.0.0.0:55555->55555/tcp   torizon_lol_1

3.  log into the container that is already running, and give yourself a shell to work with:

colibri-imx8x-070212OO:~$ docker exec -it ac58687d4245 bash

and then you can run Imagemagick at the prompt on your target.

root@837f67c0ef7c:/# convert

Version: ImageMagick 6.9.11-60 Q16 aarch64 2021-01-25 https://imagemagick.org
Copyright: (C) 1999-2021 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC Modules OpenMP(4.5)
Delegates (built-in): bzlib djvu fftw fontconfig freetype heic jbig jng jp2 jpeg        lcms lqr ltdl lzma openexr pangocairo png tiff webp wmf x xml zlib
Usage: convert-im6.q16 [options ...] file [ [options ...] file ...] [options ...       ] file

Image Settings:
  -adjoin              join images into a single multi-image file
  -affine matrix       affine transform matrix
  -alpha option        activate, deactivate, reset, or set the alpha channel
  -antialias           remove pixel-aliasing
  -authenticate password
                       decipher image with this password
  -attenuate value     lessen (or intensify) when adding noise to an image

**Reply by niels-bosman** [reply]
@HBL001 Please note that this does not install latest versions.
